lib/ansible/modules/cloud/azure/azure_rm_acr.py

```diff
@@ -0,0 +1,362 @@
+                            resource_group_name=self.resource_group,
+                            registry_name=self.name,
+                            registry_update_parameters=RegistryUpdateParameters(
+                                sku=Sku(
+                                    name=self.sku
+                                ),
+                                tags=self.tags,
+                                admin_user_enabled=self.admin_user_enabled
+                            )
+                        )
+                    else:
+                        raise Exception("Update registry failed as registry '" + self.name + "' doesn't exist.")
+                response = self.get_poller_result(poller)
+                if self.admin_user_enabled:
+                    credentials = self.containerregistry_mgmt_client.registries.list_credentials(self.resource_group, self.name)
+                else:
+                    self.log('Cannot perform credential operations as admin user is disabled')
+                    credentials = None
+            else:
+                response = None
+                credentials = None
+        except (CloudError, Exception) as exc:
+            self.log('Error attempting to create / update the ACR instance.')
+            self.fail("Error creating / updating the ACR instance: {0}".format(str(exc)))
+        return create_acr_dict(response, credentials)
+
+    def delete_acr(self):
+        '''
+        Deletes the specified container registry in the specified subscription and resource group.
+
+        :return: True
+        '''
+        self.log("Deleting the ACR instance {0}".format(self.name))
+        try:
+            self.containerregistry_mgmt_client.registries.delete(self.resource_group, self.name).wait()
+        except CloudError as e:
+            self.log('Error attempting to delete the ACR instance.')
+            self.fail("Error deleting the ACR instance: {0}".format(str(e)))
+
+        return True
+
+    def get_acr(self):
+        '''
+        Gets the properties of the specified container registry.
+
+        :return: deserialized ACR state dictionary
+        '''
+        self.log("Checking if the ACR instance {0} is present".format(self.name))
+        found = False
+        try:
+            response = self.containerregistry_mgmt_client.registries.get(self.resource_group, self.name)
+            found = True
+            self.log("Response : {0}".format(response))
+            self.log("ACR instance : {0} found".format(response.name))
+        except CloudError as e:
+            self.log('Did not find the ACR instance: {0}'.format(str(e)))
+            response = None
+        if found is True and self.admin_user_enabled is True:
+            try:
+                credentials = self.containerregistry_mgmt_client.registries.list_credentials(self.resource_group, self.name)
+            except CloudError as e:
+                self.log('List registry credentials failed: {0}'.format(str(e)))
```

Comment:
This should probably be a fatal error, since we know there *should* be credentials...